src/main/java/com/github/wovnio/wovnjava/UrlPath.java

```diff
@@ -0,0 +1,49 @@
+package com.github.wovnio.wovnjava;
+
+final class UrlPath {
+    private UrlPath() {}
+
+    public static String removeFile(String path) {
+        if (path.endsWith("/")) {
+            return path;
+        } else {
+            int index = path.lastIndexOf("/");
+            if (index > 0) {
+                return path.substring(0, index + 1);
+            } else {
+                return "/";
+            }
+        }
+    }
+
+    public static String join(String left, String right) {
+        boolean l = left.endsWith("/");
+        boolean r = right.startsWith("/");
+        if (l && r) {
+            return left + right.substring(1);
+        } else if (l || r) {
+            return left + right;
+        } else {
+            return left + "/" + right;
+        }
+    }
+
+    public static String normalize(String path) {
+        path = replaceRepeat(path, "/./", "/");            // remove redundant relative path
```

Comment:
Probably would look to replacing these functions with the Java URI class at some point. Fairly sure this relative path code isn't quite right